### PR TITLE
pkg: add table router

### PR DESCRIPTION
**Pull request by IANTHEREAL**
add table router to rename schema/table to expected schema/table. we can use it in sharding table merge scenario. @csuzhangxc @holys PTAL

### CODE REVIEW

File: pkg/table-router/router.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	SchemaPattern string `json:"schema-pattern" toml:"schema-pattern"`
+	TablePattern  string `json:"table-pattern" toml:"table-pattern"`
+	TargetSchema  string `json:"target-schema" toml:"target-schema"`
+	TargetTable   string `json:"target-table" toml:"target-table"`
+}
+
+// Valid checks validity of rule
+func (t *TableRule) Valid() error {
+	if len(t.SchemaPattern) == 0 {
+		return errors.Errorf("schema pattern of table route rule %+v should not be empty")
```
Comment (csuzhangxc) [inline pkg/table-router/router.go]: args for `%+v` not exists

File: pkg/table-router/router.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	}
+
+	if len(t.TargetSchema) == 0 {
+		return errors.Errorf("target schema of table route rule %+v should not be empty")
```
Comment (csuzhangxc) [inline pkg/table-router/router.go]: ditto

File: pkg/table-router/router.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+		targetSchema string
+		targetTable  string
+	)
+	if len(table) == 0 || len(tableRules) == 0 {
+		if len(schemaRules) == 0 {
+			return "", "", errors.NotFoundf("route %s/%s", schema, table)
+		}
+
+		if len(schemaRules) > 1 {
+			return "", "", errors.NotSupportedf("route %s/%s to rule set(%d)", schema, table, len(rules))
```
Comment (csuzhangxc) [inline pkg/table-router/router.go]: `return "", "", errors.NotSupportedf("route %s/%s to rule set(%d)", schema, table, len(schemaRules))` ? maybe not a problem

File: pkg/table-router/router_test.go [hunk 4]
```diff
@@ -87,8 +87,9 @@ func (t *testRouterSuite) TestRoute(c *C) {
 	cases = cases[1:]
 
 	// mismatched
-	_, _, err = router.Route("test_3_a", "")
-	c.Assert(err, NotNil)
+	schema, _, err = router.Route("test_3_a", "")
+	c.Assert(err, IsNil)
+	c.Assert("test_3_a", Equals, "test_3_a")
```
Comment (csuzhangxc) [inline pkg/table-router/router_test.go]: do you mean `c.Assert(schema, Equals, "test_3_a")` ?

Comment (IANTHEREAL) [reply pkg/table-router/router_test.go]: yep, lol

**Comment by csuzhangxc** [comment]
LGTM

**Pull request merged**